ADApp/pluginSrc/NDFileHDF5AttributeDataset.cpp

```diff
@@ -185,36 +169,47 @@ asynStatus NDFileHDF5AttributeDataset::writeAttributeDataset(hdf5::When_t whenTo
       extendIndexDataSet(offsets[indexed]);
     }
     // find the data based on datatype
-    ret = ndAttr->getValue(ndAttr->getDataType(), pDatavalue, MAX_ATTRIBUTE_STRING_SIZE);
+    ret = ndAttr->getValue(ndAttr->getDataType(), pDataValueStore_[attributeBatchCount_], MAX_ATTRIBUTE_STRING_SIZE);
     if (ret == ND_ERROR) {
-      memset(pDatavalue, 0, MAX_ATTRIBUTE_STRING_SIZE);
+      memset(pDataValueStore_[attributeBatchCount_], 0, MAX_ATTRIBUTE_STRING_SIZE);
     }
+    attributeBatchCount_++;
+    writeAttributeDatasetBatch(flush);
+    
+    nextRecord_++;
+  }
+
+  return status;
+}
+
+// This function is used to write the attributes from the store
+asynStatus NDFileHDF5AttributeDataset::writeAttributeDatasetBatch(int flush)
+{
+  asynStatus status = asynSuccess;
+  // Check if it is being asked to flush or if the dataStore has hit its maximum size
+  if ((flush == 1 or attributeBatchCount_ == MAX_BATCH_SIZE) and (!isUndefined_)){
+    elementSize_[0] = attributeBatchCount_;
+    // Offset has been incremented, so is at where the stored data should end
+    offset_[0] -= (attributeBatchCount_ - 1);
     // Work with HDF5 library to select a suitable hyperslab (one element) and write the new data to it
     H5Dset_extent(dataset_, dims_);
     filespace_ = H5Dget_space(dataset_);
-
+    memspace_ = H5Screate_simple(rank_, elementSize_, NULL);
     // Select the hyperslab
     H5Sselect_hyperslab(filespace_, H5S_SELECT_SET, offset_, NULL, elementSize_, NULL);
-
-    // Write the data to the hyperslab.
-    H5Dwrite(dataset_, datatype_, memspace_, filespace_, H5P_DEFAULT, pDatavalue);
-
-
-    // Check if we are being asked to flush
-    if (flush == 1){
+    H5Dwrite(dataset_, datatype_, memspace_, filespace_, H5P_DEFAULT, pDataValueStore_);
+    if(flush ==1)
```

Comment:
```suggestion
    if(flush == 1)
```
(whitespace around operator)